src/goose/cli/session.py

```diff
@@ -140,23 +141,30 @@ def process_first_message(self) -> Optional[Message]:
             return Message.user(text=user_input.text)
         return self.exchange.messages.pop()
 
-    def single_pass(self, initial_message: str) -> None:
+    def single_pass(self, initial_message: Optional[str]) -> None:
         """
         Handles a single input message and processes a reply
         without entering a loop for additional inputs.
 
         Args:
-            initial_message (str): The initial user message to process.
+            initial_message (Optional[str]): The initial user message to process.
         """
         profile = self.profile_name or "default"
         print(f"[dim]starting session | name: [cyan]{self.name}[/]  profile: [cyan]{profile}[/]")
         print(f"[dim]saving to {self.session_file_path}")
 
-        # Process initial message
-        message = Message.user(initial_message)
+        # Check to see if there is a planned operation to perform prior to the bespoke prompt
+        if self.has_plan and len(self.exchange.messages) > 0:
```

Comment:
I might make initial_message and has_plan incompatible, checking for one or the other? 

related: if we have already loaded the plan, i don't think we need line 159, although it doesn't hurt either, more important is that we make sure we don't both add initial message and hte plan

Reply:
I wanted to leave that open to the user/plugin developer to decide. Use cases in my mind were to support:
1. User wants to run a plan then ask it a question. 
2. User wants to just run plan
3. User wants to just ask a single question

This was the only way I could find to support all 3